frontend/connection/src/types.ts

```diff
@@ -26,6 +26,14 @@ import { IAppPage } from "@streamlit/protobuf"
 
 import { ConnectionState } from "./ConnectionState"
 
+// eslint-disable-next-line
+declare global {
+  interface Window {
+    __STREAMLIT_BACKEND_URL?: string
+    __STREAMLIT_HOST_CONFIG_BACKEND_URL?: string
```

Comment:
nit: maybe add a comment here to describe why we support these window variables. 

These are also meant to be base URLs, right? Maybe add that to the names to make it a bit more explicit, e.g.:
`__STREAMLIT_BACKEND_BASE_URL` and ` __STREAMLIT_HOST_CONFIG_BASE_URL`

Reply:
Btw. we also have this in `App.tsx`:

https://github.com/streamlit/streamlit/blob/de36f41d99e9c1477045113e603fa6a246dfe3ea/frontend/app/src/App.tsx#L212-L228

Does it make sense to merge these interfaces into one?

Reply:
> These are also meant to be base URLs, right? Maybe add that to the names to make it a bit more explicit, e.g.:
__STREAMLIT_BACKEND_BASE_URL and  __STREAMLIT_HOST_CONFIG_BASE_URL

Ah, yeah I prefer the names that make it more clear that these are base URLs. Changed to those.